lib/src/utils.dart

```diff
@@ -3,13 +3,40 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:convert';
+import 'constants.dart';
 
 /// Adds additional query parameters to [url], overwriting the original
 /// parameters if a name conflict occurs.
-Uri addQueryParameters(Uri url, Map<String, String> parameters) => url.replace(
-    queryParameters: Map.from(url.queryParameters)..addAll(parameters));
+Uri addQueryParameters(Uri url, Map<String, String> parameters) =>
+    url.replace(queryParameters: Map.from(url.queryParameters)..addAll(parameters));
 
 String basicAuthHeader(String identifier, String secret) {
   var userPass = '${Uri.encodeFull(identifier)}:${Uri.encodeFull(secret)}';
   return 'Basic ${base64Encode(ascii.encode(userPass))}';
 }
+
+Map<String, String> addTrackingHeaders(
+    Map<String, String> headers, String? trackingId, Map<String, String>? additionalHeaders) {
+  if (additionalHeaders != null) {
```

Comment:
Pročn použít headers.addAll(additionalHeaders) ?

Reply:
Operace je navíc mutable, není potřeba vracet headers

Reply:
Kdyz jsem pouzil addAll, predali se tam i dalsi veci do hlavicky a rozesralo to cely request. Proto je to vyreseno takhle prasacky.

Reply:
Return type upraven na void

Reply:
Které další ? Ta pointa je, že by to mělo převzít veškeré existující hlavičky z requestu, krom teda asi Content-Type

Reply:
Jj. Tak jsem to ted zkusil znovu a jde to, ale predtim mi to zlobilo. Asi spatny test.